src/ratchet.rs

```diff
@@ -248,140 +280,68 @@ impl Ratchet {
-            medium: Hash::from(&self.medium),
-            medium_counter: self.medium_counter + 1,
-            small: Hash::from(&!self.medium),
-            small_counter: 0,
-        };
-
-        let jump_count = jumped.combined_counter() - self.combined_counter();
-        (jumped, jump_count)
+        jump_count
     }
 
-    /// Returns the next small epoch of the ratchet.
+    /// Skips the ratchet to the next medium epoch.
     ///
     /// # Examples
     ///
     /// ```
     /// use skip_ratchet::Ratchet;
     ///
     /// let mut ratchet = Ratchet::new(&mut rand::thread_rng());
-    /// let r = ratchet.next_small_epoch();
+    /// ratchet.next_medium_epoch();
     /// ```
-    pub fn next_small_epoch(&self) -> Ratchet {
-        let mut ratchet = self.clone();
-        ratchet.inc();
-        ratchet
-    }
-}
-
-impl TryFrom<String> for Ratchet {
-    type Error = RatchetErr;
-
-    fn try_from(string: String) -> Result<Self, Self::Error> {
-        if string.len() != 133 {
-            return Err(RatchetErr::BadLen(string.len()));
-        }
-
-        if &string[0..2] != RATCHET_SIGNIFIER {
-            return Err(RatchetErr::BadEncoding(string[0..2].to_string()));
-        }
-        let d = URL_SAFE_NO_PAD.decode(&string[2..])?;
-
-        let mut ratchet = Ratchet {
-            large: Hash::zero(),
-            medium: Hash::zero(),
-            small: Hash::zero(),
-            small_counter: 0,
-            medium_counter: 0,
-        };
-
-        for (i, byte) in d.iter().enumerate() {
-            match i {
-                0..=31 => ratchet.small[i] = *byte,
-                32 => ratchet.small_counter = *byte,
-                33..=64 => ratchet.medium[i - 33] = *byte,
-                65 => ratchet.medium_counter = *byte,
-                66..=97 => ratchet.large[i - 66] = *byte,
-                _ => (),
-            }
+    pub fn next_medium_epoch(&mut self) -> usize {
+        if self.medium_counter == 255 {
+            return self.next_large_epoch();
         }
 
-        Ok(ratchet)
-    }
-}
+        let count_before = self.combined_counter();
 
-impl From<&Ratchet> for String {
-    fn from(ratchet: &Ratchet) -> Self {
-        let mut b: [u8; 98] = [0; 98];
+        let medium_pre = Hash::from([], self.medium);
+        self.medium = Hash::from([], medium_pre);
+        self.small = Hash::from(self.salt, medium_pre);
 
-        for (i, byte) in ratchet.small.iter().enumerate() {
-            b[i] = *byte;
-        }
-
-        b[32] = ratchet.small_counter;
-
-        for (i, byte) in ratchet.medium.iter().enumerate() {
-            b[i + 33] = *byte;
-        }
+        self.medium_counter += 1;
+        self.small_counter = 0;
 
-        b[65] = ratchet.medium_counter;
-
-        for (i, byte) in ratchet.large.iter().enumerate() {
-            b[i + 66] = *byte;
-        }
-
-        RATCHET_SIGNIFIER.to_owned() + &URL_SAFE_NO_PAD.encode(b)
+        self.combined_counter() - count_before
```

Comment:
Am I just not seeing the small incrementer anymore? Or does that live as an `impl` on a different type?

Reply:
`next_small_epoch` and `inc` would've been aliases of each other, so I just went with `inc` only.